examples/contrib/mue/FactorMuE.py

```diff
@@ -0,0 +1,310 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+A PCA model with a MuE emission (FactorMuE). Uses the MuE package.
+"""
+
+import argparse
+import datetime
+import matplotlib.pyplot as plt
+
+import torch
+import torch.nn as nn
+from torch.nn.functional import softplus
+from torch.optim import Adam
+
+import pyro
+import pyro.distributions as dist
+
+from pyro.contrib.mue.statearrangers import profile
+from pyro.contrib.mue.variablelengthhmm import VariableLengthDiscreteHMM
+
+from pyro.infer import SVI, Trace_ELBO
+from pyro.optim import MultiStepLR
+import pyro.poutine as poutine
+
+
+class Encoder(nn.Module):
+    def __init__(self, obs_seq_length, alphabet_length, z_dim):
+        super().__init__()
+
+        self.input_size = obs_seq_length * alphabet_length
+        self.f1_mn = nn.Linear(self.input_size, z_dim)
+        self.f1_sd = nn.Linear(self.input_size, z_dim)
+
+    def forward(self, data):
+
+        data = data.reshape(-1, self.input_size)
+        z_loc = self.f1_mn(data)
+        z_scale = softplus(self.f1_sd(data))
+
+        return z_loc, z_scale
+
+
+class Decoder(nn.Module):
+    def __init__(self, latent_seq_length, alphabet_length, z_dim):
+        super().__init__()
+
+        self.latent_seq_length = latent_seq_length
+        self.alphabet_length = alphabet_length
+        self.output_size = 2 * (latent_seq_length+1) * alphabet_length
+        self.f = nn.Linear(z_dim, self.output_size)
+
+    def forward(self, z):
+
+        seq = self.f(z)
+        seq = seq.reshape([-1, 2, self.latent_seq_length+1,
+                           self.alphabet_length])
+        return seq
+
+
+class FactorMuE(nn.Module):
```

Comment:
As mentioned offline, consider moving this to something like `pyro/contrib/mue/models/factormue.py`, where `.../models` contains somewhat-reusable example models. I like how that has worked out in [pyro.contrib.epidemiology.models](http://docs.pyro.ai/en/stable/_modules/pyro/contrib/epidemiology/models.html).